src/app/infuse/InfusionFinder.tsx

```diff
@@ -0,0 +1,385 @@
+        const hasInfusables = this.props.stores.some((store) =>
+          store.items.some((i) => Boolean(isInfusable(item, i)))
+        );
+        if (hasInfusables) {
+          this.setState({
+            query: item,
+            source: undefined,
+            target: item,
+            direction: Direction.INFUSE,
+            height: undefined,
+            filter: ''
+          });
+        } else {
+          this.setState({
+            query: item,
+            source: item,
+            target: undefined,
+            direction: Direction.FUEL,
+            height: undefined,
+            filter: ''
+          });
+        }
+      })
+    );
+    this.unregisterTransitionHook = router.transitionService.onBefore({}, () => this.onClose());
+
+    if (this.itemContainer.current) {
+      this.setState({ height: this.itemContainer.current.clientHeight });
+    }
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.destinyVersion !== 1 && this.props.destinyVersion === 1 && !this.state.defs) {
+      getDefinitions().then((defs) => this.setState({ defs }));
+    }
+
+    if (this.itemContainer.current && !this.state.height) {
+      this.setState({ height: this.itemContainer.current.clientHeight });
+    }
+  }
+
+  componentWillUnmount() {
+    this.subscriptions.unsubscribe();
+    if (this.unregisterTransitionHook) {
+      this.unregisterTransitionHook();
+      this.unregisterTransitionHook = undefined;
+    }
+  }
+
+  render() {
+    const { stores, searchConfig, filters } = this.props;
+    const { query, direction, filter, height } = this.state;
+    let { target, source } = this.state;
+
+    if (!query) {
+      return null;
+    }
+
+    const filterFn = filters.filterFunction(filter);
+
+    let items = _.flatMap(stores, (store) =>
+      store.items.filter(
+        (item) =>
+          (direction === Direction.INFUSE ? isInfusable(query, item) : isInfusable(item, query)) &&
+          filterFn(item)
+      )
+    );
+
+    const dupes = items.filter((item) => item.hash === query.hash);
+    dupes.sort(itemComparator);
+    items = items.filter((item) => item.hash !== query.hash);
+    items.sort(itemComparator);
+
+    target = target || items[0];
+    source = source || items[0];
+
+    let result: DimItem | undefined;
+    if (source && target && source.primStat && target.primStat) {
+      const infused = source.primStat ? source.primStat.value : 0;
+      result = copy(target);
+      (result as any).primStat.value = infused;
+    }
+
+    const missingItem = (
+      <div className="item missingItem">
+        <div className="item-img">
+          <AppIcon icon={helpIcon} />
+        </div>
+        <div className="item-stat">???</div>
+      </div>
+    );
+
+    const header = ({ onClose }: { onClose(): void }) => (
+      <div className="infuseHeader">
+        <h1>
+          {t(direction === Direction.INFUSE ? 'Infusion.InfuseTarget' : 'Infusion.InfuseSource', {
```

Comment:
Add comment {/* t('Infusion.InfuseTarget') t('Infusion.InfuseSource') /*}
or change code to
{direction === Direction.INFUSE ? t('Infusion.InfuseTarget', {name: query.name}) : t('Infusion.InfuseSource', {name: query.name})}
for detection of translate strings by #3570